.travis.yml

```diff
@@ -40,55 +40,127 @@ cache:
-    - if [[ $deps = high && $TRAVIS_BRANCH = master ]]; then SYMFONY_VERSION=$(git ls-remote --heads | grep -o '/[1-9].*' | tail -n 1 | sed s/.//); else SYMFONY_VERSION=$(cat composer.json | grep '^ *"dev-master". *"[1-9]' | grep -o '[0-9.]*'); fi
-    - if [[ $deps = high && $TRAVIS_BRANCH = master ]]; then git fetch origin $SYMFONY_VERSION; git checkout -m FETCH_HEAD; COMPONENTS=$(find src/Symfony -mindepth 3 -type f -name phpunit.xml.dist -printf '%h\n'); fi
-    # Legacy tests are skipped when deps=high and when the current branch version has not the same major version number than the next one
-    - if [[ $deps = high && ${SYMFONY_VERSION%.*} != $(git show $(git ls-remote --heads | grep -FA1 /$SYMFONY_VERSION | tail -n 1):composer.json | grep '^ *"dev-master". *"[1-9]' | grep -o '[0-9]*' | head -n 1) ]]; then LEGACY=,legacy; fi
-    - export COMPOSER_ROOT_VERSION=$SYMFONY_VERSION.x-dev
-    - if [[ ! $skip && $deps ]]; then export SYMFONY_DEPRECATIONS_HELPER=weak; fi
-    - if [[ ! $skip && $deps ]]; then mv composer.json.phpunit composer.json; fi
-    - if [[ ! $skip ]]; then composer update --no-suggest; fi
+    - |
+      # Create local composer packages for each patched components and reference them in composer.json files when cross-testing components
+      if [[ ! $deps ]]; then
+          php .github/build-packages.php HEAD^ src/Symfony/Bridge/PhpUnit
+      elif [[ ! $skip ]]; then
+          export SYMFONY_DEPRECATIONS_HELPER=weak &&
+          cp composer.json composer.json.orig &&
+          echo -e '{\n"require":{'"$(grep phpunit-bridge composer.json)"'"php":"*"},"minimum-stability":"dev"}' > composer.json &&
+          php .github/build-packages.php HEAD^ $COMPONENTS &&
+          mv composer.json composer.json.phpunit &&
+          mv composer.json.orig composer.json
+      fi
+
+    - |
+      # For the master branch, when deps=high, the version before master is checked out and tested with the locally patched components
+      if [[ $deps = high && $TRAVIS_BRANCH = master ]]; then
+          SYMFONY_VERSION=$(git ls-remote --heads | grep -o '/[1-9].*' | tail -n 1 | sed s/.//) &&
+          git fetch origin $SYMFONY_VERSION &&
+          git checkout -m FETCH_HEAD &&
+          COMPONENTS=$(find src/Symfony -mindepth 3 -type f -name phpunit.xml.dist -printf '%h\n')
+      elif [[ ! $skip ]]; then
+          SYMFONY_VERSION=$(cat composer.json | grep '^ *"dev-master". *"[1-9]' | grep -o '[0-9.]*')
+      fi
+
+    - |
+      # Legacy tests are skipped when deps=high and when the current branch version has not the same major version number than the next one
+      [[ $deps = high && ${SYMFONY_VERSION%.*} != $(git show $(git ls-remote --heads | grep -FA1 /$SYMFONY_VERSION | tail -n 1):composer.json | grep '^ *"dev-master". *"[1-9]' | grep -o '[0-9]*' | head -n 1) ]] && LEGACY=,legacy
+
+      export COMPOSER_ROOT_VERSION=$SYMFONY_VERSION.x-dev
+      if [[ ! $skip && $deps ]]; then mv composer.json.phpunit composer.json; fi
```

Comment:
do we actually need this one ?  our PHPUnit wrapper already handles the installation of the bridge, not need to force again the usage of  the one requiring it